h2o-py/h2o/model/model_base.py

```diff
@@ -1054,51 +1076,129 @@ def partial_plot(self, data, cols, destination_key=None, nbins=20, weight_column
+                    ax.set_xlabel(colPairs[0])
+                    ax.set_xlim(min(x), max(x))
+                    ax.set_ylabel(colPairs[1])
+                    ax.set_ylim(min(y), max(y))
+                    ax.set_zlim(min([min(zupper), min(zlower), min(pp[2])]), max([max(zupper), max(zlower), max(pp[2])]))
+                    ax.set_zlabel('Partial dependence')
+                    titles = '2D partial dependence plot for '+colPairs[0] + ' and '+colPairs[1]
+                    ax.set_title(titles)
+                else:  # plot 1D pdp
+                    col = cols[i]
+                    cat = data[col].isfactor()[0]
+                    upper = [a + b for a, b in zip(pp[1], pp[2]) ]  # pp[1] is mean, pp[2] is std
+                    lower = [a - b for a, b in zip(pp[1], pp[2]) ]
+                    axs = fig.add_subplot(gxs[i])
+                    self.setAxs1D(axs, upper, lower, plot_stddev, cat, pp, 0, col)  # setup graph, axis, labels and ...
+
+            if totFig > 1:
                 fig.tight_layout(pad=0.4, w_pad=0.5, h_pad=1.0)
             if(save_to_file is not None):
                 plt.savefig(save_to_file)
 
         return pps
-
+    
+    # change x, y, z to be 2-D numpy arrays in order to plot it.
+    # note that, x stays at one value for the duration of y value changes.
+    def predFor3D(self, x, y, z, colPairs, nbins, user_cols, user_num_splits):
+        # deal with y axis first
+        import numpy as np
+        ycol = colPairs[1]
+        nBins = nbins
+        if ycol in user_cols:
+            ind = user_cols.index(ycol)
+            nBins = user_num_splits[ind]
+        nrow = int(len(x)/nBins)
+        X = np.transpose(np.array(x).reshape(nrow, nBins))
+        Y = np.transpose(np.array(y).reshape(nrow, nBins))
+        Z = np.transpose(np.array(z).reshape(nrow, nBins))
+        return X,Y,Z
+    
+    def grabValues(self, pp, index, data, col, axs):
+        cat = data[col].isfactor()[0]
+        if cat:
+            labels = pp[index]
+            uniqueL =list(set(labels))
+            x = range(len(uniqueL))
+            xlab = [None]*len(uniqueL)
+            for ind in range(len(uniqueL)):
+                xlab[ind] = labels[labels.index(uniqueL[ind])]
+
+            # replace string enum labels with integer values
+            xext = [None]*len(labels)
+            for ind in range(len(labels)):
+                xext[ind] = labels.index(labels[ind])
+                
+            if index==0:    # x-axis
+                axs.set_xticks(x)
+                axs.set_xticklabels(xlab)
+            else:   # y-axis
+                axs.set_yticks(x)
+                axs.set_yticklabels(labels)
+            axs.margins(0.2) 
+
+            return xext
+        else:
+            return pp[index]
+        
+    def setAxs1D(self, axs, upper, lower, plot_stddev, cat, pp, pp_start_index, col):
```

Comment:
Is this a private function? If so, better to append with `__`

Reply:
Yeap, fixed! Thanks.